tests/Functional/Driver/OCI8/ConnectionTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\OCI8;
+
+use Doctrine\DBAL\DriverManager;
+use Doctrine\DBAL\Exception\DriverException;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+use Throwable;
+
+class ConnectionTest extends FunctionalTestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        if (TestUtil::isDriverOneOf('oci8', 'pdo_oci')) {
+            return;
+        }
+
+        self::markTestSkipped('This test requires the oci8 or pdo_oci driver.');
+    }
+
+    public function testPrepareThrowsErrorOnConnectionLost(): void
+    {
+        $this->markConnectionNotReusable();
+        $this->killCurrentSession();
+
+        $this->expectException(DriverException::class);
+
+        $this->connection->executeQuery('SELECT * FROM 1');
+    }
+
+    /**
+     * Kill the current session, by using another connection
+     * Oracle doesn't allow you to terminate the current session, so we use a second connection
+     */
+    private function killCurrentSession(): void
+    {
+        $row = $this->connection->fetchNumeric(<<<'SQL'
+        SELECT SID, SERIAL#
+        FROM V$SESSION
+        WHERE AUDSID = USERENV('SESSIONID')
+SQL);
+
+        self::assertNotFalse($row);
+        [$sid, $serialNumber] = [$row[0] ?? null, $row[1] ?? null];
```

Comment:
Why isn't `[$sid, $serialNumber] = $row` sufficient here? If the row is not `false`, then it will be an array of exactly two elements.

Reply:
Psalm flags that as an issue, and as there is only 8 suppress comments across the codebase, I went with that.

`
ERROR: PossiblyUndefinedArrayOffset - tests/Functional/Driver/OCI8/ConnectionTest.php:47:10 - Possibly undefined array key (see https://psalm.dev/167)
        [$sid, $serialNumber] = $row;
`

suppress added